eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -18,15 +18,29 @@ jobs:
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
   displayName: Generate Job Matrix
   steps:
+    - checkout: none
+
+    - pwsh: |
+        git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name).git .
+        git sparse-checkout init
+        if ("${{ parameters.AdditionalParameters.ServiceDirectory }}") {
```

Comment:
I thought you were going to condition this entire block on this parameter. Because we essentially need to revert back to the default full clone if it isn't set. 

Also keep in mind part of the reason we need to fallback to the default is for cases where we are using this on a non-public repository.

Reply:
Ah yes, forgot about that case this morning. I'll wrap both steps in a service directory conditional.

Reply:
Wouldn't a non-public repository still set ServiceDirectory?